### PR TITLE
[Routing] Add matched and default parameters to redirect responses

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18012, #19037 |
| License | MIT |
| Doc PR |  |

Finished #18012 and #19037

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
deps=high failure will be fixed when 3.4 will be merged into master

File: src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php [hunk 1]
```diff
@@ -351,33 +375,17 @@ private function compileRoute(Route $route, $name, $supportsRedirections, $paren
...
-                implode(', ', $vars),
-                str_replace("\n", '', var_export($route->getDefaults(), true))
-            );
-        } elseif ($route->getDefaults()) {
-            $code .= sprintf("            return %s;\n", str_replace("\n", '', var_export(array_replace($route->getDefaults(), array('_route' => $name)), true)));
+        if ($hasTrailingSlash || $schemes) {
+            $code .= "            return \$ret;\n";
         } else {
-            $code .= sprintf("            return array('_route' => '%s');\n", $name);
+            $code = substr_replace($code, 'return', $retOffset + 12, 6);
```
Comment (Tobion) [inline src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: this $retOffset is hard to understand. please add a comment what it does

Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: thanks, addressed

**Review by Tobion — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @artursvonda @Tobion

**Pull request merged**